Source/WebKit/WebProcess/WebPage/WebCookieJar.cpp

```diff
@@ -217,11 +217,29 @@ bool WebCookieJar::cookiesEnabled(const Document& document) const
     if (!webFrame || !webFrame->page())
         return false;
 
-    ApplyTrackingPrevention dummy;
-    if (shouldBlockCookies(webFrame.get(), document.firstPartyForCookies(), document.cookieURL(), dummy))
+    ApplyTrackingPrevention applyTrackingPreventionInNetworkProcess = ApplyTrackingPrevention::No;
+    if (shouldBlockCookies(webFrame.get(), document.firstPartyForCookies(), document.cookieURL(), applyTrackingPreventionInNetworkProcess))
         return false;
 
-    return WebProcess::singleton().ensureNetworkProcessConnection().cookiesEnabled();
+    if (applyTrackingPreventionInNetworkProcess == ApplyTrackingPrevention::No)
+        return true;
+
+    return document.cachedCookiesEnabled();
```

Comment:
What if `cachedCookiesEnabled()` hasn't been initialized yet? Since it is updated asynchronously via async IPC, it seems like we may return an incorrect result temporarily, which may lead to flakiness.

Reply:
I've made cachedCookiesEnabled() return an optional, and added a check for it so it logs error when the value is not cached.
I am not sure whether it's a common case that cookiesEnabled is checked before it's cached, since cookieURL seems to be set very early in the loading stage. So currently if the value is not cached yet, we return true (note the subsequent cookie getting/setting will still fail as they will send message to network process).
If this error is popular, we might consider adding sync message for cookiesEnabled later. WDYT?